rpcs3/Emu/Cell/Modules/cellRtc.cpp

```diff
@@ -3,298 +3,1672 @@
+	pszDateTime[0xc] = hourDigits[1];
+	pszDateTime[0xd] = ':';
+
+	//	Minutes - 2020-04-13T10:XX:31.35+66:40
+	char minDigits[3];
+	sprintf(minDigits, "%02hi", u16{date_time->minute});
+	pszDateTime[0xe]  = minDigits[0];
+	pszDateTime[0xf]  = minDigits[1];
+	pszDateTime[0x10] = ':';
+
+	//	Seconds - 2020-04-13T10:56:XX.35+66:40
+	char secDigits[3];
+	sprintf(secDigits, "%02hi", u16{date_time->second});
+	pszDateTime[0x11] = secDigits[0];
+	pszDateTime[0x12] = secDigits[1];
+	pszDateTime[0x13] = '.';
+
+	// Microseconds - 2020-04-13T10:56:31.XX+66:40
+	char microDigits[3];
+	sprintf(microDigits, "%02u", u32{date_time->microsecond});
+	pszDateTime[0x14] = microDigits[0];
+	pszDateTime[0x15] = microDigits[1];
+
+	if (iTimeZone == 0)
+	{
+		pszDateTime[0x16] = 'Z';
+		pszDateTime[0x17] = '\0';
+	}
+	else
+	{
+		if (iTimeZone < 0)
+		{
+			tzone             = -tzone;
+			pszDateTime[0x16] = '-';
+		}
+		else
+		{
+			pszDateTime[0x16] = '+';
+		}
+		u32 uVar1 = tzone >> 0x1f;
+
+		u32 lVar9 = (tzone / 0x3c + uVar1) - uVar1;
+		tzone -= (lVar9 << 6) - (lVar9 << 2);
+		uVar1             = tzone >> 0x1f;
+		u32 lVar11        = (lVar9 / 10 + (lVar9 >> 0x1f)) - (lVar9 >> 0x1f);
+		u32 lVar8         = (tzone / 10 + uVar1) - uVar1;
+		pszDateTime[0x17] = lVar11 + (lVar11 / 10) * -10 + '0';
+		pszDateTime[0x18] = (lVar9 - ((lVar11 << 1) + (lVar11 << 3))) + '0';
+		pszDateTime[0x19] = ':';
+		pszDateTime[0x1a] = lVar8 + (lVar8 / 10) * -10 + '0';
+		pszDateTime[0x1b] = (tzone - ((lVar8 << 1) + (lVar8 << 3))) + '0';
+		pszDateTime[0x1c] = '\0';
+	}
+
+	//	TODO;
 	return CELL_OK;
 }
 
-s32 cellRtcFormatRfc3339LocalTime(vm::ptr<char> pszDateTime, vm::ptr<CellRtcTick> pUtc)
+error_code cellRtcFormatRfc3339LocalTime(vm::ptr<char> pszDateTime, vm::cptr<CellRtcTick> pUtc)
 {
 	cellRtc.todo("cellRtcFormatRfc3339LocalTime(pszDateTime=*0x%x, pUtc=*0x%x)", pszDateTime, pUtc);
 
-	return CELL_OK;
+	vm::var<sys_page_attr_t> page_attr;
+
+	error_code ret = sys_memory_get_page_attribute(pszDateTime.addr(), page_attr);
+	if (ret != CELL_OK)
+	{
+		return CELL_RTC_ERROR_INVALID_POINTER;
+	}
+
+	ret = sys_memory_get_page_attribute(pUtc.addr(), page_attr);
+	if (ret != CELL_OK)
+	{
+		return CELL_RTC_ERROR_INVALID_POINTER;
+	}
+
+	vm::var<s32> timezone;
+	vm::var<s32> summertime;
+
+	ret = sys_time_get_timezone(timezone, summertime);
+	if (ret != CELL_OK)
+	{
+		return ret;
+	}
+
+	return cellRtcFormatRfc3339(pszDateTime, pUtc, *timezone + *summertime);
 }
 
-s32 cellRtcParseDateTime(vm::ptr<CellRtcTick> pUtc, vm::cptr<char> pszDateTime)
+/*
+ Takes a RFC2822 / RFC3339 / asctime String, and converts it to a CellRtcTick
+*/
+error_code cellRtcParseDateTime(vm::ptr<CellRtcTick> pUtc, vm::cptr<char> pszDateTime)
 {
-	cellRtc.todo("cellRtcParseDateTime(pUtc=*0x%x, pszDateTime=%s)", pUtc, pszDateTime);
+	cellRtc.error("cellRtcParseDateTime(pUtc=*0x%x, pszDateTime=%s) -- Implement me", pUtc, pszDateTime);
+
+	std::tm t = {};
+	std::string tz;
+
+
+
+
+	//Below code kinda works
```

Comment:
We always use a single space. There's probably more like this in this file
```suggestion
	// Below code kinda works
```